topic-operator/src/main/java/io/strimzi/operator/topic/v2/TopicOperatorConfig.java

```diff
@@ -0,0 +1,247 @@
+                kafkaClientProps.put(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, this.keystoreLocation());
+                kafkaClientProps.put(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, this.keystorePassword());
+            }
+        }
+
+        if (this.saslEnabled()) {
+            putSaslConfigs(kafkaClientProps);
+        }
+        return kafkaClientProps;
+    }
+
+    private void putSaslConfigs(Map<String, Object> kafkaClientProps) {
+        TopicOperatorConfig config = this;
+        String saslMechanism;
+        String jaasConfig;
+        String username = config.saslUsername();
+        String password = config.saslPassword();
+        String configSaslMechanism = config.saslMechanism();
+
+        if (username.isEmpty() || password.isEmpty()) {
+            throw new InvalidConfigurationException("SASL credentials are not set");
+        }
+
+        if ("plain".equals(configSaslMechanism)) {
+            saslMechanism = "PLAIN";
+            jaasConfig = "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"" + username + "\" password=\"" + password + "\";";
+        } else if ("scram-sha-256".equals(configSaslMechanism) || "scram-sha-512".equals(configSaslMechanism)) {
+            jaasConfig = "org.apache.kafka.common.security.scram.ScramLoginModule required username=\"" + username + "\" password=\"" + password + "\";";
+
+            if ("scram-sha-256".equals(configSaslMechanism)) {
+                saslMechanism = "SCRAM-SHA-256";
+            } else {
+                saslMechanism = "SCRAM-SHA-512";
+            }
+        } else {
+            throw new IllegalArgumentException("Invalid SASL_MECHANISM type: " + configSaslMechanism);
+        }
+
+        kafkaClientProps.put(SaslConfigs.SASL_MECHANISM, saslMechanism);
+        kafkaClientProps.put(SaslConfigs.SASL_JAAS_CONFIG, jaasConfig);
+    }
+
+    @Override
+    public String toString() {
+        String mask = "********";
+        return "TopicOperatorConfig{" +
+                "\n\tnamespace='" + namespace + '\'' +
+                "\n\tlabelSelector=" + labelSelector +
+                "\n\tbootstrapServers='" + bootstrapServers + '\'' +
+                "\n\tclientId='" + clientId + '\'' +
+                "\n\tfullReconciliationIntervalMs=" + fullReconciliationIntervalMs +
+                "\n\ttlsEnabled=" + tlsEnabled +
+                "\n\ttruststoreLocation='" + truststoreLocation + '\'' +
+                "\n\ttruststorePassword='" + mask + '\'' +
+                "\n\tkeystoreLocation='" + keystoreLocation + '\'' +
+                "\n\tkeystorePassword='" + mask + '\'' +
+                "\n\tsslEndpointIdentificationAlgorithm='" + sslEndpointIdentificationAlgorithm + '\'' +
+                "\n\tsaslEnabled=" + saslEnabled +
+                "\n\tsaslMechanism='" + saslMechanism + '\'' +
+                "\n\tsaslUsername='" + saslUsername + '\'' +
+                "\n\tsaslPassword='" + mask + '\'' +
```

Comment:
what's the purpose of having a `mask` variable to be concatenated with other corresponding "password" related fields instead of having just the asterisks in the string and avoid concatenation (which allocate a new String each time)?